### PR TITLE
Higher res splash screen for >1080p resolutions

### LINKED ISSUE #23814: Workbench: Use high resolution splash screen
On 4K displays the splash screen for the workbench is very small. We need to use the high resolution version from the images folder: [MantidSplashScreen_2400x1758.jpg](/mantidproject/mantid/blob/master/images/MantidSplashScreen_2400x1758.jpg).

This code concerning this lives in [mainwindow.py](https://github.com/mantidproject/mantid/blob/master/qt/applications/workbench/workbench/app/mainwindow.py#L87).

### PR DESCRIPTION
**Pull request by dtasev**
**Description of work.**
Adds a new, larger splash image.

Adds a condition to check the screen's width, follows these rules:

| Width | Splash |
|---------|---------|
|1920 | Old |
| 2048 | Old |
| 2560 | New |
| 3840 | New |

These are the widths from the default resolutions up to 3840x2160 for a screen with 1.777 aspect ratio.

**To test:**
- Whether it works properly with multiple displays

Fixes #23814
Before:
image
After:
image

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 1]
```diff
@@ -82,13 +82,19 @@ def qapplication():
...
 # Importing resources loads the data in
 from workbench.app.resources import qCleanupResources  # noqa
-atexit.register(qCleanupResources)
 
-SPLASH = QSplashScreen(QPixmap(':/images/MantidSplashScreen.png'),
-                       Qt.WindowStaysOnTopHint)
+atexit.register(qCleanupResources)
+width = QApplication.desktop().availableGeometry().width()
+
+if width > 2048:
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: As the only variable is the image name is it worth not duplicating the `QSplashScreen` call and defining a function to return the appropriate image?

**Comment by martyngigg** [comment]
@samueljackson92 If you have a second would you be able to check this on your large macOS screen?

File: qt/applications/workbench/workbench/app/mainwindow.py [hunk 2]
```diff
@@ -82,13 +82,23 @@ def qapplication():
 # -----------------------------------------------------------------------------
 # Importing resources loads the data in
 from workbench.app.resources import qCleanupResources  # noqa
+
 atexit.register(qCleanupResources)
 
-SPLASH = QSplashScreen(QPixmap(':/images/MantidSplashScreen.png'),
+
+def _get_splash_image_name():
```
Comment (martyngigg) [inline qt/applications/workbench/workbench/app/mainwindow.py]: :+1:

**Comment by martyngigg** [comment]
It works on my dual monitors but they are not 4K. Do we need to try and test a dual-4K setup?

**Comment by DTasev** [comment]
I don't think it's necessary, but I guess it wouldn't hurt if we have access to such a setup - just to make sure. 

The issue I was worried about was whether `availableGeometry` would give back the total width of all screen, or just the one where the Workbench window is at, as per [this](http://doc.qt.io/qt-5/qdesktopwidget.html#screen-geometry). 

> QDesktopWidget also inherits the QWidget properties, width() and height(), which specify the size of the desktop. However, for desktops with multiple screens, the size of the desktop is the union of all the screen sizes, so width() and height() should not be used for computing the size of a widget to be placed on one of the screens.

Maybe it would be more robust to do use the [primary screen information](http://doc.qt.io/qt-5/qguiapplication.html#primaryScreen-prop), becoming `QGuiApplication().primaryScreen().screenGeometry().width()`

**Comment by martyngigg** [comment]
I think your last suggestion makes sense, especially if you have 1 4K and 1 non-4K display? You want it to give you the correct image for the screen it is on I think.

**Review by martyngigg — APPROVED** [review]
This works for me on dual non-4K displays. I think checking the primary screen is the correct thing to do to avoid problems with different resolutions.

**Comment by AndreiSavici** [comment]
The broken test is now fixed on master

**Pull request merged**